framework/Samplers/DynamicEventTree.py

```diff
@@ -183,7 +183,27 @@ def localFinalizeActualSampling(self,jobObject,model,myInput,genRunQueue=True):
     # Read the branch info from the parent calculation (just ended calculation)
     # This function stores the information in the dictionary 'self.actualBranchInfo'
     # If no branch info, this history is concluded => return
-    if not self.__readBranchInfo(jobObject.output, jobObject.getWorkingDir()):
+
+    ## There are two ways to get at the working directory from the job instance
+    ## and both feel a bit hacky and fragile to changes in the Runner classes.
+    ## They are both listed below and the second inevitably stems from the first.
+    ## I am wondering if there is a reason we cannot use the model.workingDir
+    ## from above here? Granted the job instance should have a snapshot of
+    ## whatever the model's current working directory was for that evaluation,
+    ## and it could have changed in the meantime, so I will keep this as is for
+    ## now, but note this should be re-evaluated in the future. -- DPM 4/12/17
+    # codeModel = jobObject.args[0]
+    # jobWorkingDir = codeModel.workingDir
```

Comment:
Even if the ExternalRunner was kept in the code, my suggestion would be to have it called inside the Code model which means that an InternalRunner would still be wrapped around it, and every jobObject here would be an instance of an InternalRunner and not an ExternalRunner.